### DISCUSSION TITLE
Porous_flow simulation with Segmentation fault: 11

### DISCUSSION THREAD

**Discussion by wjin33** [idaholab/moose#17503] [Q&A Modules: General] [answered]
Hi, 

I am using Porous Flow/Falcon to run brine flow and heat transfer problems. Prepared input file with water as the fluid runs ok, yet, switched to brine as the injection/extraction files have this error:
Segmentation fault: 11

Please help. 

Many thanks!

Attached is the input file, I have to change the file type to be able to upload. 

[Cranfield_HT_season_caps_brine_newboundary_coarse.log](https://github.com/idaholab/moose/files/6252919/Cranfield_HT_season_caps_brine_newboundary_coarse.log)
input.

**Comment by WilkAndy** [comment, marked as answer]
Your input file uses an exodus mesh, so i can't be exactly sure what your problem is, but i believe your problem is in your `PorousFlowUnsaturated` block.  Instead of `fp = brine`, you need to specify
```
  fluid_properties_type = PorousFlowBrine
  nacl_name = f_nacl
```
See the doco at https://mooseframework.inl.gov/source/actions/PorousFlowFullySaturated.html .

There are other things you might like to change about your input file.  Eg, the `f_nacl` is mass fraction, so setting it as 0.62 means you're simulating with 62% of salt.  You'll probably also want to use `PorousFlowFullySaturated`, as that is more efficient and suited to your case (above doco).

When you use these Actions, you do not need to supply any `FluidProperties`, since PorousFlow does it all for you (see the "Materials Added" section of the above documentation).  If you like, you can supply a custom `fluid_properties_type` if the ones automatically built by PorousFlow are too slow or don't suit you for some reason.

a

**Reply by WilkAndy** [reply]
Did this work, @wjin33 ?

**Reply by wjin33** [reply]
No, it does not work, @WilkAndy. I tried both the fullysaturated action and the fluid-type lines. Before posting the question here, I have already tried different actions and input lines for brine. All failed. 

I combined both the exodus mesh file and the input file here. Please kindly take a look. 

[Archive.zip](https://github.com/idaholab/moose/files/6266549/Archive.zip)

**Reply by WilkAndy** [reply]
[Cranfield_HT_season_caps_brine_newboundary_coarse.txt](https://github.com/idaholab/moose/files/6267580/Cranfield_HT_season_caps_brine_newboundary_coarse.txt)

Hi @wjin33 , i wonder whether you need to update MOOSE.  Your input file works fine for me, after i've removed the `EnthalpySink` stuff, which i don't have in standard `porous_flow`.  So there could be something wrong with the `EnthalpySink`.  The input file that works is found attached (it doesn't converge well, but that's probably due to the missing `EnthalpySink`).

a

**Reply by WilkAndy** [reply]
Did updating MOOSE work, @wjin33 ?

**Reply by wjin33** [reply]
Dear Andy @WilkAndy, I have already updated the MOOSE, and it works for water with my kernel "EnthalpSink". Upon your suggestions, I removed the  EnthalpSink with PorousFlowPolyLineSink, and I can simulate the brine without error. Many thanks. 

I double-checked my kernel and attached them for your reference. I think there are two aspects that may contribute to the  error: 
- I am using the SinglePhaseFluidProperties to calculate the enthalpy injection, which works fine for water97 equation of state.  Yet, for brine, this is not properly implemented.
- I did not calculate the Jacobian for this kernel. 

Any comments are appreciated.

Thanks,

Wencheng

[EnthalpySink.zip](https://github.com/idaholab/moose/files/6298296/EnthalpySink.zip)

**Reply by WilkAndy** [reply]
Is `EnthalpySink` trying to accomplish the same thing as @andrsd #17235 ?   If so, just take something like https://www.ratebeer.com/Ratings/Beer/Beer-Ratings.asp?BeerID=84035 and sit with him until it's done.
